node/src/main/scala/com/wavesplatform/state/StateSnapshot.scala

```diff
@@ -172,23 +163,28 @@ object StateSnapshot {
         .toMap
 
     val leaseStates: Map[ByteStr, LeaseDetails] =
-      pbSnapshot.leaseStates
-        .map(ls =>
-          ls.leaseId.toByteStr -> LeaseDetails(
-            ls.sender.toPublicKey,
-            PBRecipients.toAddress(ls.recipient.toByteArray, AddressScheme.current.chainId).explicitGet(),
-            ls.amount,
-            ls.status match {
-              case TransactionStateSnapshot.LeaseState.Status.Cancelled(c) =>
-                LeaseDetails.Status.Cancelled(c.height, if (c.transactionId.isEmpty) None else Some(c.transactionId.toByteStr))
-              case _ =>
-                LeaseDetails.Status.Active
-            },
-            ls.originTransactionId.toByteStr,
-            ls.height
-          )
-        )
-        .toMap
+      pbSnapshot.leaseStates.map { ls =>
+        ls.status match {
+          case TransactionStateSnapshot.LeaseState.Status.Active(value) =>
+            ls.leaseId.toByteStr -> LeaseDetails(
+              value.sender.toPublicKey,
+              value.recipient.toAddress,
+              value.amount,
+              LeaseDetails.Status.Active,
+              sourceId = ByteStr.empty,
+              height = 0
+            )
+          case _: TransactionStateSnapshot.LeaseState.Status.Cancelled | TransactionStateSnapshot.LeaseState.Status.Empty =>
+            ls.leaseId.toByteStr -> LeaseDetails(
+              PublicKey(ByteStr.fill(KeyLength)(0)),
+              Address(Array.fill(Address.HashLength)(0)),
+              0,
+              LeaseDetails.Status.Cancelled(0, None),
+              sourceId = ByteStr.empty,
+              height = 0
```

Comment:
As @phearnot said, we can fill the data here from the snapshot

Reply:
Done